mermithid/processors/misc/ComplexLineShape.py

```diff
@@ -0,0 +1,598 @@
+    # shake off spectrum for the ith state
+    def spectrum1(self, i, E):
+        if self.E_b_Scale[i] == 0:
+            spectrum = self.I(i, E)
+        else:
+            factor = np.arctan(2*(self.Ecore - E - self.B_Binding[i]+self.epsilon_shake_spectrum)/self.Gamma_Width[i])/(np.pi)+0.5
+            spectrum = self.A_Intensity[i]*factor*self.P_1s_nprime(self.E_b_Scale[i], self.Ecore-E-self.B_Binding[i]+self.epsilon_shake_spectrum)
+        return spectrum
+
+    # adding up shake spectrum for all the states
+    def full_shake_spectrum(self, E, start_number_of_i, end_number_of_i):
+        full_spectrum = 0
+        for i in range(start_number_of_i, end_number_of_i, 1):
+            full_spectrum += self.spectrum1(i, E)
+        return full_spectrum
+
+    # shake spectrum by adding up shake spectrum for all the states up to i=24
+    def shake_spectrum(self):
+        x_array = self.input_std_eV_array
+        x_array = flip_array(x_array)
+        shake_spectrum = self.full_shake_spectrum(x_array, 0, 24)
+        return shake_spectrum
+    ###############################################################################
+
+class ComplexLineShape(BaseProcessor):
+
+    def InternalConfigure(self, params):
+        '''
+        Configure
+        '''
+        # Read other parameters
+        self.bins_choice = reader.read_param(params, 'bins_choice', [])
+        self.gases = reader.read_param(params, 'gases', ["H2","Kr"])
+        self.max_scatters = reader.read_param(params, 'max_scatters', 20)
+        self.max_comprehensive_scatters = reader.read_param(params, 'max_comprehensive_scatters', 4)
+        # This is an important parameter which determines how finely resolved
+        # the scatter calculations are. 10000 seems to produce a stable fit, with minimal slowdown
+        self.num_points_in_std_array = reader.read_param(params, 'num_points_in_std_array', 10000)
+        self.RF_ROI_MIN = reader.read_param(params, 'RF_ROI_MIN', 25850000000.0)
+        self.B_field = reader.read_param(params, 'B_field', 0.957810722501)
+        self.shake_spectrum_parameters_json_path = reader.read_param(params, 'shake_spectrum_parameters_json_path', 'shake_spectrum_parameters.json')
+        self.path_to_osc_strengths_files = reader.read_param(params, 'path_to_osc_strengths_files', '/host/')
+
+    def InternalRun(self):
+
+        # Read shake parameters from JSON file
+        self.shakeSpectrumClassInstance = ShakeSpectrumClass(self.shake_spectrum_parameters_json_path, self.std_eV_array())
+
+        # number_of_events = len(self.data['StartFrequency'])
+        # self.results = number_of_events
+
+        a = self.data['StartFrequency']
+
+        # fit with shake spectrum
+        data_hist_freq, freq_bins= np.histogram(a,bins=self.bins_choice)
+        histogram = data_hist_freq
+        bins = freq_bins
+        guess = np.where(np.array(histogram) == np.max(histogram))[0][0]
+        print(guess)
```

Comment:
if you want to print something use one of the logger functions, eg logger.info()